core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1683,8 +1686,48 @@ class KafkaApis(val requestChannel: RequestChannel,
     // If this is considered to leak information about the broker version a workaround is to use SSL
     // with client authentication which is performed at an earlier stage of the connection where the
     // ApiVersionRequest is not available.
+    val apiVersionRequest = request.body[ApiVersionsRequest]
+
+    // flush unofficial client set every hour
+    if (System.currentTimeMillis() - flushStartTime > unofficialClientSetTtl) {
+      unofficialClientIdentities.clear()
+      flushStartTime = System.currentTimeMillis()
+    }
+
+    if (config.unofficialClientLoggingEnable) {
+      val softwareName = apiVersionRequest.data.clientSoftwareName()
+      val clientIdentity = request.context.clientId() + "-" + request.context.clientAddress() + "-" + request.context.principal()
+      val UNOFFICIAL_CLIENT = "UNOFFICIAL"
+
+      def getClientTypeFromClientSoftwareName(softwareName: String): String = {
```

Comment:
we only intend to log unofficial clients, so why do we care about which type of clients like avro/raw/admin? is it possible if we just check if (softwareName == defaultName) -> unofficial clients?

Reply:
It seems that there are four default names: "apache-kafka-java", "linkedin-kafka-java", "li-oss-consumer-java", and "li-oss-producer-java". Based on that would it still be better to check for each of these than the current way?